lib/puppet/http/client.rb

```diff
@@ -11,11 +11,10 @@ def initialize(pool: Puppet::Network::HTTP::Pool.new, ssl_context: nil, system_s
     @default_system_ssl_context = system_ssl_context
     @redirector = Puppet::HTTP::Redirector.new(redirect_limit)
     @retry_after_handler = Puppet::HTTP::RetryAfterHandler.new(retry_limit, Puppet[:runinterval])
-    @resolvers = build_resolvers
   end
 
   def create_session
-    Puppet::HTTP::Session.new(self, @resolvers)
+    Puppet::HTTP::Session.new(self, build_resolvers)
```

Comment:
Would it be worth adding a test to ensure each session has a unique set of resolvers? Or would that just be pedantic?

Reply:
Updated